### PR TITLE
Add support for plugin config

### PR DESCRIPTION
**Pull request by rogeliog**
Depends on https://github.com/facebook/jest/pull/6603

### CODE REVIEW

**Comment by SimenB** [comment]
Available in Jest 23.3.0, so should be good to go 🙂

File: src/file_name_plugin.js [hunk 1]
```diff
@@ -5,23 +5,32 @@ import FileNamePatternPrompt, {
...
   _stdin: stream$Readable | tty$ReadStream;
   _stdout: stream$Writable | tty$WriteStream;
   _prompt: Prompt;
   _projects: SearchSources;
+  _config: PluginConfig;
 
   constructor({
     stdin,
     stdout,
+    config,
```
Comment (SimenB) [inline src/file_name_plugin.js]: should have a default value

File: src/file_name_plugin.js [hunk 2]
```diff
@@ -48,8 +57,8 @@ class FileNamePlugin {
   // eslint-disable-next-line class-methods-use-this
   getUsageInfo() {
     return {
-      key: 'p',
-      prompt: 'filter by a filename regex pattern',
+      key: this._config.key || 'p',
```
Comment (SimenB) [inline src/file_name_plugin.js]: check in constructor and assign there?

Comment (rogeliog) [reply src/file_name_plugin.js]: Sounds like a good idea? Do you mean assigning the `this._key` and `this._prompt` as instance variables?

Comment (SimenB) [reply src/file_name_plugin.js]: Yes 🙂

**Pull request merged**